design-docs/handling-encodings.md

```diff
@@ -0,0 +1,87 @@
+# Encoding
+
+Updated: __2016/03/19__
+Status: __Under Review__
+
+This document describes the rules and conventions for handling the character encoding of file contents in general, and file names 
+and comments in archives. It applies to tasks which need to read, write and transform text files, as well as tasks which create
+archives (zip files, tar files, etc.)
+
+## Use Cases
+
+### File Content Encoding
+
+Any time we need to generate a text file, or to read, transform and write the contents of a text file, we need to know what 
+character encoding(s) to use.
+
+Common situations include:
+
+* Copying a file from one location to another.
+* Creating an archive (zip, tar, etc.).
+* Transforming text in a file.
+
+[GRADLE-1267](https://issues.gradle.org/browse/GRADLE-1267) describes just one instance of where this functionality is needed.
+
+### File Name Encoding
+
+Each file system has an idea of what character sets are used to represent files in the filesystem, and it's important to be 
+able to use filenames in archives that are encoded using the encoding of the target file system, which is not necessarily the same
+as the file system where the archive is created.
+
+This use case is why the Zip task and the War task have an `encoding` option added in 
+[Pull Request 499](https://github.com/gradle/gradle/pull/499) which allows creating a zip archive which contains files whose names 
+are encoded with a character encoding different from the default platform encoding.
+
+## Solution
+
+This design generally proposes that any task which needs to read and write text or encode file names in an archive, should support two configurable settings:
+
+1. `contentEncoding` - The encoding to be used to convert bytes to characters (when reading), and characters to bytes (when writing).
+2. `fileNameEncoding` - The encoding to be used for names of files in an archive.
+
+Both properties default to platform default encoding if they are not specified in the task's configuration block.
```

Comment:
I wouldn't use that property. First because it seems to be specific to the Sun/Oracle JVM. Second because it's not what is used by default by the standard zip classes of the JDK. But what is used by default is UTF-8, actually, so I should change the plan and say that UTF-8 is used by default.
BTW, the current documentation seems to be incorrect: it says the platform default encoding is used, but reading the code, it's not: no encoding is set on the ZipOutputStream, which thus uses UTF-8.